includes/abstracts/abstract-wc-stripe-payment-gateway.php

```diff
@@ -113,7 +113,7 @@ public function is_no_such_source_error( $error ) {
 		return (
 			$error &&
 			'invalid_request_error' === $error->type &&
-			preg_match( '/No such source/i', $error->message )
+			preg_match( '/(No such source|No such payment)/i', $error->message )
```

Comment:
This seems a bit more precise to me in terms of what we are expecting from the error message:

```suggestion
			preg_match( '/No such (source|PaymentMethod)/i', $error->message )
```

What do you think?